addons/image/freeimage.c

```diff
@@ -0,0 +1,168 @@
+/* Allegro wrapper routines for FreeImage
+ * by Karthik Kumar Viswanathan <karthikkumar@gmail.com>.
+ */
+
+#include <FreeImage.h>
+
+#include "allegro5/allegro.h"
+#include "allegro5/allegro_image.h"
+#include "allegro5/internal/aintern_exitfunc.h"
+#include "allegro5/internal/aintern_image.h"
+
+#include "iio.h"
+
+ALLEGRO_DEBUG_CHANNEL("image")
+
+static bool freeimage_initialized = false;
+
+bool _al_init_fi(void)
+{
+   if(freeimage_initialized)
+       return true;
+   FreeImage_Initialise(FALSE);
+   _al_add_exit_func(_al_shutdown_fi, "_al_shutdown_fi");
+   return true;
+}
+
+void _al_shutdown_fi(void)
+{
+   if(!freeimage_initialized)
+       return;
+   FreeImage_DeInitialise();
+   freeimage_initialized = false;
+}
+
+static ALLEGRO_BITMAP *_al_fi_to_al_bitmap(FIBITMAP *fib) {
+   int width = 0, height = 0;
+   ALLEGRO_BITMAP *bitmap = NULL;
+   width = FreeImage_GetWidth(fib);
+   height = FreeImage_GetHeight(fib);
+   bitmap = al_create_bitmap(width, height);
+   if(bitmap) {
```

Comment:
In Allegro we add a space after the `if`, `for` etc. Would you mind doing a search/replace for all such cases in these files?

Reply:
Done!